frontend/src/pages/voting/index.tsx

```diff
@@ -104,6 +104,8 @@ export const VotingPage = () => {
     ) !== -1;
 
   const isVotingClosed = data && !data.conference.isVotingOpen;
+  const userCannotVote =
+    loggedIn && (loading || (cannotVoteErrors ?? false) || (error ?? false));
```

Comment:
This line looks very complex

Can you describe in words what you are trying to achieve?

Reply:
I would like to reuse this condition to understand whether I have to show the filters or not... In order to not have the ugly thing that the filter show and disappear immediately when user cannot vote

Reply:
Agree is quite complex, took me some time to figure out the right condition... :D

Reply:
So I am having a look

cannotVoteErrors:

can be changed to

```typescript
  const cannotVoteErrors =
    (error?.graphQLErrors ?? []).findIndex(
      (e) => e.message === "You need to have a ticket to see submissions",
    ) !== -1;
```

so you don't need `(cannotVoteErrors ?? false)`, cannotVoteErrors will always be true.

Note that I think this would have worked without `??` maybe by doing `!!cannotVoteErrors` so it goes from undefined to boolean :)

Reply:
Another thing, why do we `error ?? false`? why do we check error here? We already kind of do it in `cannotVoteErrors` and if I have to be honest, if the error the user is seeing is not that they can't vote, I wouldn't want to display this message

Reply:
Oki, I see why we do `error ?? false` because we want this variable to be false when the error is not related to the voting permissions

This feels very confusing  🤔